core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadWriteLDAPUserStoreConstants.java

```diff
@@ -143,7 +143,7 @@ public class ReadWriteLDAPUserStoreConstants {
         setProperty(UserStoreConfigConstants.usernameJavaRegEx, "Username RegEx (Java)", "[a-zA-Z0-9._-|//]{3,30}$",
                 UserStoreConfigConstants.usernameJavaRegExDescription,
                 new Property[] { USER.getProperty(), STRING.getProperty(), TRUE.getProperty() });
-        setProperty(UserStoreConfigConstants.usernameJavaScriptRegEx, "Username RegEx (Javascript)", "^[\\S]{3,30}$",
+        setProperty(UserStoreConfigConstants.usernameJavaScriptRegEx, "Username RegEx (Javascript)", "^[^\\s*?%]{3,30}$",
```

Comment:
Do we know if these restricted characters are not working for these userstore type as well?

Reply:
Set UsernameJavaScriptRegEx to be same as UsernameJavaRegEx by https://github.com/wso2/carbon-kernel/pull/3776/commits/e3332fa009c93f8373733685604a8dca70b83071